### PR TITLE
[MM-40632] Updates to the Developer Documentation and README, remove unnecessary files and code

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This PR makes changes to the documentation within the desktop repo. Updates to development processes, links and refreshes images.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40632

```release-note
NONE
```

### CODE REVIEW

**Comment by devinbinnie** [comment]
Added a whole bunch of reviewers, not sure who needs to look at this but wanted to make sure I didn't miss anyone.
Feel free to remove yourself if needed.

Also, some of the links will depend on changes made here: https://github.com/mattermost/mattermost-developer-documentation/pull/973

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,39 +1,46 @@
...
 If the problem you encounter can be reproduced on web browsers, it may be an issue with Mattermost server (or Chrome).
 
+If this is the case, please create an issue in the [mattermost-server](https://github.com/mattermost/mattermost-server) or [mattermost-webapp](https://github.com/mattermost/mattermost-webapp) repositories.
+
 ### Try "Clear Cache and Reload"
-It's available as `Ctrl(Command) + Shift + R`.
-Some layout problems are caused by browser cache.
-Especially, this kind of issue might happen when you have updated Mattermost server.
+Sometimes issues can be resolved simply by refreshing your Mattermost server within the app.  
+You can do this by pressing `Cmd/Ctrl+Shift+R` in the Mattermost Desktop App, or you can go to the menu > View > Clear Cache and Reload
```
Comment (cwarnermm) [inline CONTRIBUTING.md:17]:
```suggestion
You can do this by pressing `CMD/CTRL+SHIFT+R` in the Mattermost Desktop App, or you can go to the menu and select **View > Clear Cache and Reload**.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,39 +1,46 @@
...
-
-For example:
+If the issue still persists, please provide detailed information to help us to understand the problem. Include information such as:
 * How to reproduce, step-by-step
 * Expected behavior (or what is wrong)
 * Screenshots (for GUI issues)
-* Application version
-* Operating system
-* Mattermost server version
+* Desktop App application version (can be viewed by going to 3-dot menu > Help, or Top Menu > Mattermost > About Mattermost on macOS)
```
Comment (cwarnermm) [inline CONTRIBUTING.md:24]:
```suggestion
* Desktop App version (can be viewed by going to 3-dot menu > Help, or **Menu > Mattermost > About Mattermost** on macOS).
```

File: README.md [hunk 3]
```diff
@@ -4,37 +4,29 @@ Native desktop application for [Mattermost](http://www.mattermost.org/) running
...
-1. Download a file from the [downloads page](https://about.mattermost.com/downloads).
-2. Launch `Mattermost` in the unarchived folder.
-3. On the first launch, please input name and URL for your Mattermost server. For example, `myserver : https://mattermost.example.com`.
+Detailed guides are available at [docs.mattermost.com](https://docs.mattermost.com/install/desktop-app-install.html).
 
-### Quit
-Ctrl or Command + Q to quit.
+1. Download a file from the [downloads page](https://mattermost.com/download/#mattermostApps) or from the [releases page](https://github.com/mattermost/desktop/releases).
+2. Run the installer or unzip the archive.
+3. Launch Mattermost from your Applications folder, menu or the unarchived folder
```
Comment (cwarnermm) [inline README.md:28]:
```suggestion
3. Launch Mattermost from your Applications folder, menu, or the unarchived folder.
```

File: README.md [hunk 4]
```diff
@@ -4,37 +4,29 @@ Native desktop application for [Mattermost](http://www.mattermost.org/) running
...
+3. On the first launch, please input a name and URL for your Mattermost server. For example, `myserver : https://mattermost.example.com`.
```
Comment (cwarnermm) [inline README.md:29]:
```suggestion
3. On the first launch, please enter a name and URL for your Mattermost server. For example, `myserver : https://mattermost.example.com`.
```

File: TESTING.md [hunk 5]
```diff
@@ -8,21 +13,21 @@ New bug reports benefiting the next release will be documented in the release no
...
+1. If you haven't already, create an account on https://community.mattermost.com/
  - Set your username to be the same as your GitHub username
 
 2. Install the latest Mattermost Desktop App
- - Download the latest Mattermost Desktop App from the [Mattermost Download page](https://about.mattermost.com/download/#mattermostApps)
- - Follow the [Desktop Application Install Guides](https://about.mattermost.com/default-desktop-app-install-documentation/) to install the app for your platform
- - Use the [Desktop Application's User Guide](https://about.mattermost.com/default-desktop-app-documentation/) to add https://pre-release.mattermost.com/core as a new team
+ - Download the latest pre-release Mattermost Desktop App from the [GitHub Releases page](https://github.com/mattermost/desktop/releases)
+ - Follow the [Desktop Application Install Guides](https://docs.mattermost.com/install/desktop-app-install.html) to install the app for your platform
+ - Use the [Managing Servers Guide](https://docs.mattermost.com/messaging/managing-desktop-app-servers.html) to add https://community.mattermost.com/core as a new server
```
Comment (cwarnermm) [inline TESTING.md:22]:
```suggestion
 - Use the [Managing Servers Guide](https://docs.mattermost.com/messaging/managing-desktop-app-servers.html) to add https://community.mattermost.com/core as a new server.
```

File: TESTING.md [hunk 6]
```diff
@@ -8,21 +13,21 @@ New bug reports benefiting the next release will be documented in the release no
...
 
-1. If you haven't already, create an account on https://pre-release.mattermost.com/core
+1. If you haven't already, create an account on https://community.mattermost.com/
  - Set your username to be the same as your GitHub username
 
 2. Install the latest Mattermost Desktop App
- - Download the latest Mattermost Desktop App from the [Mattermost Download page](https://about.mattermost.com/download/#mattermostApps)
- - Follow the [Desktop Application Install Guides](https://about.mattermost.com/default-desktop-app-install-documentation/) to install the app for your platform
- - Use the [Desktop Application's User Guide](https://about.mattermost.com/default-desktop-app-documentation/) to add https://pre-release.mattermost.com/core as a new team
+ - Download the latest pre-release Mattermost Desktop App from the [GitHub Releases page](https://github.com/mattermost/desktop/releases)
```
Comment (cwarnermm) [inline TESTING.md:20]:
```suggestion
 - Download the latest pre-release Mattermost Desktop App from the [GitHub Releases page](https://github.com/mattermost/desktop/releases).
```

File: TESTING.md [hunk 7]
```diff
@@ -8,21 +13,21 @@ New bug reports benefiting the next release will be documented in the release no
...
+ - Follow the [Desktop Application Install Guides](https://docs.mattermost.com/install/desktop-app-install.html) to install the app for your platform
```
Comment (cwarnermm) [inline TESTING.md:21]:
```suggestion
 - Follow the [Desktop Application Install Guides](https://docs.mattermost.com/install/desktop-app-install.html) to install the app for your platform.
```

File: TESTING.md (hunk 5 above)
Comment (cwarnermm) [inline TESTING.md:22]:
```suggestion
 - Use the [Managing Servers Guide](https://docs.mattermost.com/messaging/managing-desktop-app-servers.html) to add https://community.mattermost.com/core as a new server.
```

File: TESTING.md [hunk 8]
```diff
@@ -8,21 +13,21 @@ New bug reports benefiting the next release will be documented in the release no
...
+ - Use the [Managing Servers Guide](https://docs.mattermost.com/messaging/managing-desktop-app-servers.html) to add https://community.mattermost.com/core as a new server 
  - Hit "Save" and log in
```
Comment (cwarnermm) [inline TESTING.md:23]:
```suggestion
 - Select "Save" and log in to Mattermost.
```

File: TESTING.md [hunk 9]
```diff
@@ -8,21 +13,21 @@ New bug reports benefiting the next release will be documented in the release no
...
 
-3. Go to the [Public Test Channel](https://pre-release.mattermost.com/core/channels/public-test-channel) and try the following:
- - Post a message with information on what you're testing, for example: `Testing Mattermost Desktop App 3.4.1 on Windows 10 64-bit`
+3. Go to the [Public Test Channel](https://community.mattermost.com/core/channels/public-test-channel) and try the following:
+ - Post a message with information on what you're testing, for example: `Testing Mattermost Desktop App 5.0.2 on Windows 10 64-bit`
```
Comment (cwarnermm) [inline TESTING.md:26]:
```suggestion
 - Post a message with information on what you're testing, for example: `Testing Mattermost Desktop App 5.0.2 on Windows 10 64-bit`.
```

**Review by cwarnermm — COMMENTED** [review]
Great updates! I've provided feedback inline.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,39 +1,46 @@
...
-Some layout problems are caused by browser cache.
-Especially, this kind of issue might happen when you have updated Mattermost server.
+Sometimes issues can be resolved simply by refreshing your Mattermost server within the app.  
+You can do this by pressing `Cmd/Ctrl+Shift+R` in the Mattermost Desktop App, or you can go to the menu > View > Clear Cache and Reload
 
 ### Write detailed information
-Detailed information is very helpful to understand the problem.
-
-For example:
+If the issue still persists, please provide detailed information to help us to understand the problem. Include information such as:
```
Comment (Willyfrog) [inline CONTRIBUTING.md:20]: we should throw here a link to either community or the issues from the gihub repo.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,39 +1,46 @@
...
-* Operating system
-* Mattermost server version
+* Desktop App application version (can be viewed by going to 3-dot menu > Help, or Top Menu > Mattermost > About Mattermost on macOS)
+* Operating System
+* Mattermost Server version
 
 ## Feature idea
-Please see http://www.mattermost.org/feature-requests/ .
+If you have an idea for a new feature, we'd love to hear about it!  
+Please let us know in the Mattermost Community server by making a post in the [Feature Proposals](https://community-daily.mattermost.com/core/channels/feature-ideas) channel.
```
Comment (Willyfrog) [inline CONTRIBUTING.md:30]: is feature proposals still the way to go? we have many different places where people can ask for features that I'm a bit lost (that channel, user voice, github discussions and probably I'm missing more)

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,39 +1,46 @@
...
-Pull requests are welcome. Thank you for your great work!
-
-1. When you edit the code, please confirm `npm test` successfully finishes.
-2. Please update `docs/*.md` if it's necessary.
-3. In the description of your pull request, please include:
-   * Operating System version on which you tested
-   * Mattermost server version on which you tested
-   * New or updated unit tests for your changes
-4. Please complete the [Mattermost CLA](http://www.mattermost.org/mattermost-contributor-agreement/) prior to submitting a PR.
+If you are interesting on working on an issue, we would very much appreciate your help!
```
Comment (Willyfrog) [inline CONTRIBUTING.md:33]:
```suggestion
If you are interested on working on an issue, we would very much appreciate your help!
```

File: TESTING.md [hunk 13]
```diff
@@ -8,21 +13,21 @@ New bug reports benefiting the next release will be documented in the release no
 
 To contribute to the process of testing the Mattermost Desktop App:
 
-1. If you haven't already, create an account on https://pre-release.mattermost.com/core
+1. If you haven't already, create an account on https://community.mattermost.com/
```
Comment (Willyfrog) [inline TESTING.md:16]:
```suggestion
1. If you haven't already, create an account on our [Community Server](https://community.mattermost.com/)
```

**Review by Willyfrog — APPROVED** [review]
added a few suggestions, but looking good :)

File: CONTRIBUTING.md (hunk 11 above)
Comment (devinbinnie) [reply CONTRIBUTING.md:30]: Not sure, but I know putting them in GitHub is generally not the best way. I think directly people to Community in general makes the most sense, but I'm not completely sure it's the best way.

File: CONTRIBUTING.md (hunk 10 above)
Comment (devinbinnie) [reply CONTRIBUTING.md:20]: What's the link here for? For examples of tickets?

**Review by cwarnermm — APPROVED** [review]

**Review by aspleenic — APPROVED** [review]

File: docs/release-process.md [hunk 14]
```diff
@@ -1,177 +0,0 @@
-# Release Process
```
Comment (amyblais) [inline docs/release-process.md:1]: Do we want to consider moving this to the Handbook together with other release processes https://handbook.mattermost.com/operations/research-and-development/product/release-process?

Comment (devinbinnie) [reply docs/release-process.md:1]: I'd be okay with that, it was in here before and I was wondering if this belonged internally or not.

Comment (amyblais) [reply docs/release-process.md:1]: I opened a PR here: https://github.com/mattermost/mattermost-handbook/pull/925.

**Review by amyblais — APPROVED** [review]

File: docs/release-process.md (hunk 14 above)
Comment (devinbinnie) [reply docs/release-process.md:1]: Great, I'll remove this file from the repo

File: README.md [hunk 15]
```diff
@@ -4,37 +4,29 @@ Native desktop application for [Mattermost](http://www.mattermost.org/) running
 
 Originally created as "electron-mattermost" by Yuya Ochiai. Developed using [Electron](http://electron.atom.io/).
```
Comment (BenLloydPearson) [inline README.md:5]: Can you please add the following befor this link?

[Mattermost](https://mattermost.com) is an open source platform for secure collaboration across the entire software development lifecycle. This repo is for the native desktop application that's built on [Electron](http://electron.atom.io/); it runs on Windows, Mac, and Linux.

File: README.md [hunk 16]
```diff
@@ -4,37 +4,29 @@ Native desktop application for [Mattermost](http://www.mattermost.org/) running
...
-2. Launch `Mattermost` in the unarchived folder.
-3. On the first launch, please input name and URL for your Mattermost server. For example, `myserver : https://mattermost.example.com`.
+Detailed guides are available at [docs.mattermost.com](https://docs.mattermost.com/install/desktop-app-install.html).
 
-### Quit
-Ctrl or Command + Q to quit.
+1. Download a file from the [downloads page](https://mattermost.com/download/#mattermostApps) or from the [releases page](https://github.com/mattermost/desktop/releases).
+2. Run the installer or unzip the archive.
+3. Launch Mattermost from your Applications folder, menu, or the unarchived folder.
+3. On the first launch, please enter a name and URL for your Mattermost server. For example, `myserver : https://mattermost.example.com`.
```
Comment (BenLloydPearson) [inline README.md:29]: We should remove "myserver :" from this. See [here](https://github.com/mattermost/desktop/pull/1802#discussion_r727789116)

**Review by BenLloydPearson — APPROVED** [review]
I have two small revision requests, but this looks great otherwise.

File: CONTRIBUTING.md (hunk 11 above)
Comment (BenLloydPearson) [reply CONTRIBUTING.md:30]: You can use this URL: https://handbook.mattermost.com/contributors/contributors/ways-to-contribute#propose-new-features

It's pretty light on content, but this is something we should maintain in a single location.

**Pull request merged**